File: SourceAFIS/Extraction/Minutiae/MinutiaShuffler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SourceAFIS.General;

namespace SourceAFIS.Extraction.Minutiae
{
    public static class MinutiaShuffler
    {
        public static void Shuffle(FingerprintTemplate template)
        {
            int seed = 0;
            foreach (var minutia in template.Minutiae)
                seed += minutia.Direction + minutia.Position.X + minutia.Position.Y + (int)minutia.Type;
            template.Minutiae = Calc.Shuffle(template.Minutiae, new Random(seed)).ToList();
        }
    }
}

File: java/sourceafis/src/sourceafis/extraction/minutiae/MinutiaShuffler.java
/**
 * @author Veaceslav Dubenco
 * @since 19.10.2012
 */
package sourceafis.extraction.minutiae;

import java.util.Random;

import sourceafis.general.Calc;
import sourceafis.templates.Minutia;
import sourceafis.templates.TemplateBuilder;

/**
 * 
 */
public final class MinutiaShuffler {
	public void Shuffle(TemplateBuilder template) {
		int seed = 0;
		for (Minutia minutia : template.minutiae)
			seed += minutia.Direction + minutia.Position.X + minutia.Position.Y
					+ minutia.Type.getValue();
		template.minutiae = Calc.Shuffle(template.minutiae, new Random(seed));
	}

}
